### PR TITLE
cmake: linker_script: Fixes for `CMAKE_LINKER_GENERATOR`

### LINKED ISSUE #75497: zbus: orphan sections when building with `CONFIG_CMAKE_LINKER_GENERATOR=y`
**Describe the bug**
`zbus` makes heavily use of iterable sections, however warnings are printed when building for `CMAKE_LINKER_GENERATOR`, and does not pass tests. 

**To Reproduce**
```shell
west build -p -b qemu_cortex_m3 tests/subsys/zbus/unittests -- -DCONFIG_CMAKE_LINKER_GENERATOR=y
west build -t run
```

**Expected behavior**
Tests should pass.

**Impact**
Minor impact as this probably isn't used currently.

**Logs and console output**
```
[122/126] Linking C executable zephyr/zephyr_pre0.elf
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_invalid_obs_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_invalid_obs_'
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_foo2_msg_sub_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_foo2_msg_sub_'
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_foo_msg_sub_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_foo_msg_sub_'
[126/126] Linking C executable zephyr/zephyr.elf
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_invalid_obs_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_invalid_obs_'
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_foo2_msg_sub_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_foo2_msg_sub_'
/opt/zephyr-sdk-0.16.5-1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `._k_fifo.static._zbus_observer_fifo_foo_msg_sub_' from `app/libapp.a(main.c.obj)' being placed in section `._k_fifo.static._zbus_observer_fifo_foo_msg_sub_'
```

**Environment (please complete the following information):**
 - OS: Ubuntu 22.04
 - Toolchain: Zephyr SDK 0.16.5-1
 - Commit SHA: 34e4edc28a3806885821d863cc3ebc9c60f5b817

### PR DESCRIPTION
**Pull request by pdgendt**
Add missing section for `k_fifo` kernel objects and fix `zbus` sections.

Fixes: #75497

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by rodrigopex — APPROVED** [review]

File: cmake/linker_script/common/common-ram.cmake [hunk 1]
```diff
@@ -96,8 +97,7 @@ if(CONFIG_ZTEST)
 endif()
 
 if(CONFIG_ZBUS)
-  zephyr_iterable_section(NAME zbus_observer GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN CONFIG_LINKER_ITERABLE_SUBALIGN)
-  zephyr_iterable_section(NAME zbus_channel_observation_mask GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN CONFIG_LINKER_ITERABLE_SUBALIGN)
+  zephyr_iterable_section(NAME zbus_channel_observation_mask GROUP DATA_REGION ${XIP_ALIGN_WITH_INPUT} SUBALIGN 1)
```
Comment (rodrigopex) [inline cmake/linker_script/common/common-ram.cmake:100]: Why did you remove the `zbus_observer` section instead of fixing that?

**Review by rodrigopex — COMMENTED** [review]
Thanks for the PR. Was `zbus_observer` section wrong?

File: cmake/linker_script/common/common-ram.cmake (hunk 1 above)
Comment (pdgendt) [reply cmake/linker_script/common/common-ram.cmake:100]: It's added in `common-rom.cmake`.

Comment (rodrigopex) [reply cmake/linker_script/common/common-ram.cmake:100]: I see, sure. It is constant. So, it was wrong indeed. Thanks for that.

**Review by rodrigopex — APPROVED** [review]
